hazelcast/src/main/java/com/hazelcast/sql/impl/expression/ConstantExpression.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+import com.hazelcast.sql.impl.row.Row;
+import com.hazelcast.sql.impl.type.QueryDataType;
+import com.hazelcast.sql.impl.type.QueryDataTypeFamily;
+import com.hazelcast.sql.impl.type.converter.Converter;
+import com.hazelcast.sql.impl.type.converter.Converters;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Constant expression.
+ *
+ * @param <T> Return type.
+ */
+public final class ConstantExpression<T> implements Expression<T>, IdentifiedDataSerializable {
+
+    private QueryDataType type;
+    private T value;
+
+    public ConstantExpression() {
+        // No-op.
+    }
+
+    private ConstantExpression(T value, QueryDataType type) {
+        this.type = type;
+        this.value = value;
+    }
+
+    public static ConstantExpression<?> create(Object value, QueryDataType type) {
+        if (type.getTypeFamily() == QueryDataTypeFamily.NULL) {
```

Comment:
Minor: this if-statement could be replaced with:
```
 assert type.getTypeFamily() != QueryDataTypeFamily.NULL || value == null
```

Reply:
fixed in 07ba390b92c9954b4e3da91a5adb4be8054eb66b